internal/gcsutil/util.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package gcsutil contains utility functions for Google Cloud Storage data access.
+package gcsutil
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+// GCSClient contains a Google Cloud Storage client.
+type GCSClient struct {
+	Client *storage.Client
+}
+
+// NewClient creates and returns a new GCSClient.
+func NewClient(ctx context.Context) (*GCSClient, error) {
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not create a new Google Cloud Storage client: %v", err)
+	}
+	defer client.Close()
+	gcsClient := GCSClient{
+		Client: client,
+	}
+	return &gcsClient, nil
+}
+
+// ListBlobs returns all the objects paths in a Google Cloud Storage bucket
+// under a given relative path.
+func (client *GCSClient) ListBlobs(ctx context.Context, bucketName string, relativePath string) ([]string, error) {
```

Comment:
Make `ctx` a field in `GCSClient` and replace `ctx` with `client.ctx` below. 

Also Golang style guide prefers shorter names: go/go-style/decisions#receiver-names. So perhaps replace `client` with `gc`.

Reply:
I renamed it to `c`.

Reply:
> Make ctx a field in GCSClient and replace ctx with client.ctx below.

I replied above :)